components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -8,399 +8,179 @@
-	local op1norm = op1.status == STATUS_HAS_SCORE
-	local op2norm = op2.status == STATUS_HAS_SCORE
-	if op1norm and op2norm then
-		local op1setwins = CustomMatchGroupInput._getSetWins(op1)
-		local op2setwins = CustomMatchGroupInput._getSetWins(op2)
-		if op1setwins + op2setwins > 0 then
-			return op1setwins > op2setwins
-		else
-			return tonumber(op1.score) > tonumber(op2.score)
-		end
-	else return CustomMatchGroupInput._sortOpponents(op1, op2, op1norm, op2norm) end
+CustomMatchGroupInput.processMap = FnUtil.identity
+
+---@param opponent table
+---@return table
+function CustomMatchGroupInput.getOpponentExtradata(opponent)
+	if not Logic.isNumeric(opponent.score2) then
+		return {}
+	end
+
+	return {
+		score1 = tonumber(opponent.score),
+		score2 = tonumber(opponent.score2),
+		score3 = tonumber(opponent.score3),
+		set1win = Logic.readBool(opponent.set1win),
+		set2win = Logic.readBool(opponent.set2win),
+		set3win = Logic.readBool(opponent.set3win),
+		additionalScores = true
+	}
 end
 
----@param opp table
+---@param opponent table
 ---@return integer
-function CustomMatchGroupInput._getSetWins(opp)
-	local extradata = opp.extradata or {}
+function CustomMatchGroupInput._getSetWins(opponent)
+	local extradata = opponent.extradata
 	local set1win = extradata.set1win and 1 or 0
 	local set2win = extradata.set2win and 1 or 0
 	local set3win = extradata.set3win and 1 or 0
 	return set1win + set2win + set3win
 end
 
---
--- match related functions
---
-
----@param matchArgs table
----@return {date: string, dateexact: boolean, timestamp: integer, timezoneId: string?, timezoneOffset: string?}
-function matchFunctions.readDate(matchArgs)
-	return MatchGroupInput.readDate(matchArgs.date, {'tournament_enddate'})
-end
-
 ---@param match table
 ---@return table
-function matchFunctions.getTournamentVars(match)
+function MatchFunctions.getTournamentVars(match)
 	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', '2v2'))
-	match.showh2h = Logic.emptyOr(match.showh2h, Variables.varDefault('showh2h'))
-	return MatchGroupInput.getCommonTournamentVars(match)
-end
-
----@param match table
----@return table
-function matchFunctions.getVodStuff(match)
-	match.stream = Streams.processStreams(match)
-	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
-
-	return match
+	return MatchGroupInputUtil.getCommonTournamentVars(match)
 end
 
 ---@param match table
 ---@return boolean
-function matchFunctions.isFeatured(match)
+function MatchFunctions.isFeatured(match)
 	return tonumber(match.liquipediatier) == 1
 		or tonumber(match.liquipediatier) == 2
 end
 
 ---@param match table
 ---@return table
-function matchFunctions.getExtraData(match)
+function MatchFunctions.getExtraData(match)
 	local opponent1 = match.opponent1 or {}
 	local opponent2 = match.opponent2 or {}
 
-	local showh2h = Logic.readBool(match.showh2h)
+	local showh2h = Logic.emptyOr(match.showh2h, Variables.varDefault('showh2h'))
```

Comment:
changes behavior 
what if `|showh2h=false` is set?

Reply:
Removed it entirely instead, wa unused